### PR TITLE
React Native SDK: Add session tracking with `SessionPlugin`

### PR DESCRIPTION
**Pull request by jonathanchaffer**
## Overview
Add session tracking with configurable foreground & background timeouts.

## Changes
- Introduce `SessionPlugin` + `SessionPluginHelper` to manage session state and context for events
- Add `session` object to `event.context` with the following shape:
```typescript
export type ContextSession = {
  sessionId: number; // epoch timestamp
  sessionIndex: number;
  sessionStart?: boolean; // true if this is the first event in the session, undefined if otherwise
  eventIndex: number;
  previousSessionId: number | null;
  firstEventId: string;
  firstEventTimestamp: string;
};
```
- Also add `sessionId` and `sessionStart` to the base `event.context` object for parity with the JS SDK.
- Add configurable foreground and background session timeouts, with the ability to disable session tracking altogether by setting both to 0.

## Testing
Should work the same for both iOS and Android. For iOS:
1. In `examples/AnalyticsReactNativeExample/App.tsx`, replace `<WRITE_KEY>` with an iOS source's write key
2. Run the RN example app on a simulator:
    - From repo root, run `yarn bootstrap`, then in two separate terminals run `yarn example start` and `yarn example ios`
    - Note: Xcode 26 can have issues with RN 0.72. You can add this to the `Podfile`'s post-install to fix this. Run `pod install` in the `ios` directory afterwards.
  ```
      # Xcode 16+ treats Yoga's user-defined literals as deprecated errors (RN 0.72).
    installer.pods_project.targets.each do |target|
      target.build_configurations.each do |config|
        config.build_settings['OTHER_CPLUSPLUSFLAGS'] ||= ['$(inherited)']
        config.build_settings['OTHER_CPLUSPLUSFLAGS'] << '-Wno-deprecated-literal-operator'
      end
    end
```
3. Perform actions in the example app, and verify that the events show up in the source's Debugger page with the new `session` tracking data:
  
Screenshot 2026-06-10 at 10 03 54 AM

## Summary by CodeRabbit

* **New Features**
  * Added session tracking that enriches events with session context, configurable foreground/background timeouts, automatic rotation on lifecycle changes, and persisted session state across restarts.
* **Tests**
  * Expanded test coverage for session behavior, rotation, persistence, and edge cases.

### CODE REVIEW

File: packages/core/src/plugins/session/SessionPluginHelper.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    state: SessionState | undefined,
+    now: number
+  ): SessionState | undefined {
+    if (state === undefined) {
+      return state;
+    }
+
+    return {
+      ...state,
+      backgroundedAt: now,
```
Comment (nicksmider) [inline packages/core/src/plugins/session/SessionPluginHelper.ts]: I think this should probably be 

```typescript
return { ...state, backgroundedAt: state.backgroundedAt ?? now }
```

Comment (jonathanchaffer) [reply packages/core/src/plugins/session/SessionPluginHelper.ts]: Good catch! Updated and added a test for it

**Review by nicksmider — APPROVED** [review]
